### ISSUE TITLE
Close state isn't saved properly

### ISSUE THREAD

**Issue by NhKPaNdA** [riderkick/FMD#500]
Since the .94 update that changed to SQLITE, the downloading state doesn't save correctly on close. Some downloads remain while others dissappear or even have a changed status (eg. Was finished on close but starts downloading when re-opened)

**Comment by riderkick** [comment]
Its should be saved if fmd closed properly. Enable logging, and look at fmd.log.

**Comment by taskforce22** [comment]
I'm also having this same issue, only in my case, nothing completed is getting saved between closing and reopening the program.  Only stopped/failed get saved.  I've turned on the log but it shows no issues.

**Comment by taskforce22** [comment]
I deleted downloads.db and downloadedchapters.db in works and now it is working.  Maybe it has something to do with older versions causing issues?

**Comment by riderkick** [comment]
Post your downloads.db and fmd.log

**Comment by NhKPaNdA** [comment]
Here's my fmd.log and downloads.db files.

[FMD.zip](https://github.com/riderkick/FMD/files/813069/FMD.zip)

**Comment by riderkick** [comment]
Log file looks fine. And start/stop/exit is fine too. Everything saved properly.
I guess it's because of some issue when converting from old `works.ini` to `downloads.db.`
If you still have your `works.ini`, can you post it here?

**Comment by NhKPaNdA** [comment]
Unfortunately I don't. I deleted it after I saw the downloads moved over to the downloads.db.

**Comment by taskforce22** [comment]
Here are my files. I've also provided a screencap of what should be in the db.  However, it is only saving the one entry you see in the DB.  I don't have the ini but it didn't import anything from it except the stopped/cancelled stuff from before.  I'll provide my old DB files as well.  There literally should have been at least 100 chapters or fare more imported but I got nothing completed carried over when 0.9.94 installed.  The db is from after I've closed the program showing the stuff in the screenshot.

[http://www.exiled-destiny.com/temp/fmdissue.rar](url)

**Comment by NhKPaNdA** [comment]
I updated and tried it out. It's better but still has issues. I had a downloading task and a completed task when I exited. The downloading task was stopped and on resume it started at the begging instead of continuing from the chapter it was on and the completed task dissapeared.

**Comment by SilentNN** [comment]
I've only downloaded 2 chapters since 9.96, but neither of them remained after an exit and reopen, individually for each. No chapters have stuck since the SQLite update for me.

I don't have my .ini's anymore since I renamed them back to the original after the SQLite update, then they appeared to have been converted away after reopening FMD.

I can provide a .db and .log later if you want one from me as well.
